apps/api/src/app/workflows-v2/usecases/extract-default-values-from-schema/extract-default-values-from-schema.usecase.ts

```diff
@@ -51,11 +52,24 @@ export class ExtractDefaultValuesFromSchemaUsecase {
 
         result[key] = this.getValue(value, key);
 
-        const nestedDefaults = this.extractDefaults(value);
+        const nestedDefaults = this.extractDefaults(value, controlValues);
         if (Object.keys(nestedDefaults).length > 0) {
           result[key] = { ...result[key], ...nestedDefaults };
         }
       }
+    } else if (schema.anyOf && schema.anyOf.length > 0) {
+      /*
+       * If control values do match the schema, we should return the default values from that schema.
+       * Otherwise return the default values from the first one
+       */
+      const fistSchemaMatch = schema.anyOf.find((item) => {
+        return isMatchingJsonSchema(item, controlValues);
+      });
+      if (typeof fistSchemaMatch === 'object') {
+        return this.extractDefaults(fistSchemaMatch, controlValues);
+      } else if (typeof schema.anyOf[0] === 'object') {
+        return this.extractDefaults(schema.anyOf[0], controlValues);
+      }
```

Comment:
The `digest` schema consists of the `anyOf` values. We need to match the control values with the one of the schema and return the default values.